src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -1,11 +1,33 @@
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
 import { tl } from 'app/i18next-t';
 import { getHashtagsFromNote } from 'app/inventory/note-hashtags';
+import { getDamageTypeForSubclassDef } from 'app/inventory/subclass';
 import { Loadout } from 'app/loadout-drawer/loadout-types';
 import { matchText, plainString } from 'app/search/search-filters/freeform';
+import { DestinyItemType } from 'bungie-api-ts/destiny2';
+import _ from 'lodash';
+import memoizeOne from 'memoize-one';
 import { FilterDefinition } from '../../filter-types';
 import { quoteFilterString } from '../../query-parser';
 import { LoadoutFilterContext, LoadoutSuggestionsContext } from '../loadout-filter-types';
 
+function deduplicate<T>(someArray: (T | undefined | null)[]) {
+  return _.compact(Array.from(new Set(someArray)));
+}
+
+const damageDefsByDamageType = memoizeOne((defs: D2ManifestDefinitions) =>
+  _.keyBy(Object.values(defs.DamageType.getAll()), (d) => d.enumValue),
+);
```

Comment:
You could probably share the same copy of this with the one that was in d2-item-factory, maybe by moving them to a common location?

Reply:
Making a new util as we don't have one for definitions, which was surprising tbh